test/core/base/cache.js

```diff
@@ -1,32 +1,173 @@
-describe('axe._cache', function() {
+describe('axe._cache', function () {
   'use strict';
 
-  it('should set items from the cache without error', function() {
-    function fn() {
-      axe._cache.set('foo', 'bar');
-    }
-    assert.doesNotThrow(fn);
-  });
+  describe('set', function () {
+    it('should set items without error', function () {
+      function fn() {
+        axe._cache.set('foo', 'bar');
+      }
+      assert.doesNotThrow(fn);
+    });
+
+    // undefined is a valid result type of a potentially expensive operation in axe-core
+    it('should set `undefined` without error', function () {
+      axe._cache.set('foo', undefined);
+
+      function fn() {
+        assert.isUndefined(axe._cache.get('foo'));
```

Comment:
You're testing the `get` function in the `set` describe block. We should make sure that setting `undefined` does not throw, not that `get` does not throw

```suggestion
      function fn() {
        axe._cache.set('foo', undefined);
```